torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1820,6 +2094,8 @@ def hint():
                     source == symbol_to_source[expr][0]
                 ):
                     continue
+                if is_dim(source):
+                    dim_constraints.add_equality(source, expr)
```

Comment:
It's kind of annoying that you have to remember to do a dim_constraints.add anywhere you push to exprs but I can't think way of a good way to force someone to do this

Reply:
Actually, can't we just get rid of expr entirely, and use dim_constraints to generate the guards in the end? Why can't we do this?

Reply:
I don't see any reason not to do it, other than wanting to limit the blast radius in one go. There's some parallel simplification that we could get rid of.

Reply:
OK, happy to agree this as PoR and then ship PR as is. But I don't want to turn it on "for real" without translation validation.